shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/response/GlobalResponse.kt

```diff
@@ -21,6 +21,25 @@ data class GlobalResponse(
     val stops: Map<String, Stop>,
     val trips: Map<String, Trip>,
 ) {
+    constructor(
+        objects: ObjectCollectionBuilder
+    ) : this(
+        objects.lines,
+        objects.routePatterns
+            .flatMap {
+                objects.trips[it.value.representativeTripId]?.stopIds?.map { stopId ->
+                    stopId to it.key
+                }
+                    ?: emptyList()
+            }
+            .groupBy { it.first }
+            .mapValues { it.value.map { stopAndPattern -> stopAndPattern.second } },
```

Comment:
```suggestion
            .groupBy({ it.first }, { it.second }),
```

The Kotlin standard library is [pretty nice](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.sequences/group-by.html). (The Elixir standard library [also has this](https://hexdocs.pm/elixir/Enum.html#group_by/3).)